### DISCUSSION TITLE
NvimTreeToggle open tree in tree buffer instead of closing it

### DISCUSSION THREAD

**Discussion by gdnathan** [nvim-tree/nvim-tree.lua#1106] [Q&A] [answered]
I'v used nvm-tree for a while, and i switched OS, and, with the same configuration, when i do <C-e> (i mapped this to ":NvimTreeToggle<Cr>"), it open a window on the right, with nvim-tree, and move my cursor on this window. When i do <C-e> again:
before switching OS: nvim-tree window closes itself and its window, and my cursos come back on the buffers where i was coding.
after: nevim-tree close itself, but not its window. it stays on the right, and open another file

what did i missed ?

**Comment by kyazdani42** [comment, marked as answer]
This might be related to your config not being perfectly up to date maybe. did you miss the configuration option `actions.open_file.quit_on_open` ?

**Reply by gdnathan** [reply]
It was not set on my previous config, as i did not want to, well, "quit on open", so i don't think the problem come from this

**Reply by kyazdani42** [reply]
quit on open is a bit misleading, but it seems it's doing what you are describing in your question, or maybe i understand this wrong.
What do you mean by nvim-tree closes itself but not its window ?

**Reply by gdnathan** [reply]
Ok here is more details:
<C-e> on window open nvim-tree window : OK
<C-e> on nevim-tree window act like "enter" (open directory, open files WITHIN nvim-tree window), i expect it to toggle NvimTreeToggle:

`map("<C-e>", ":NvimTreeToggle<Cr>")`

C-e is not maped anywhere else, and it acting as "enter" only happen in Nvim-tree window

**Reply by kyazdani42** [reply]
i believe `<C-e>` is mapped by default in nvim-tree now, you should override it in the setup :)

**Reply by gdnathan** [reply]
Thanks, it was it ! Works as expected ! `<C-e>` was mapped to "open in place", so i had this unexpected behavior !

Thanks for you help !
